### DISCUSSION TITLE
Signed commit - verified/unverified/none conditions

### DISCUSSION THREAD

**Discussion by ghost** [Mergifyio/mergify#3484] [Ideas]
👋🏻

Would it be possible to add conditions to assess the verification status of the commits in a PR?

https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification

I am investigating the use case to warn and possible reject PRs where a commit is `Partially verified` or `Unverified` but continue if `No verification status`.

**Comment by jd** [comment]
Yes, that sounds possible. GitHub exposes the verification status as part of the [commits API](https://docs.github.com/en/rest/reference/repos#commits).

IIUC your use case, you would add something like `#commits-unverified=0` and `#commits-partially-verified=0` to your merge conditions?

**Reply by ghost** [reply]
Yeah something like that. We'd probably do something like `#commits-unverified>0` and add some error/warning comments in the short term but, principle is the same 🤓

**Reply by jd** [reply]
Understood. We've added this on our roadmap. 😉
